File: Day-9-Object-part-4/class.js
class Vehicle {
  constructor(brand, model) {
    this.brand = brand;
    this.model = model;
  }
}

class Car extends Vehicle {}

class Truck extends Vehicle {
  constructor(brand, model, year) {
    super(brand, model);
    this.year = year;
  }
}

class Car extends Vehicle {}

class Truck extends Vehicle {
  constructor(brand, model, year) {
    super(brand, model);
    this.year = year;
  }
}

newCar = new Car("Toyota", "Corolla");

console.log(newCar.brand);
console.log(newCar.model);

newTruck = new Truck("Toyota", "Hilux", 2020);

console.log(newTruck.brand);
console.log(newTruck.model);
console.log(newTruck.year);

File: Day-9-Object-part-4/class.py
class Vehicle:
    def __init__(self, brand, model):
        self.brand = brand
        self.model = model

# No overriding parent class
class Car(Vehicle):
    pass # Inherit everything automatically

# Overriding parent class, add new attribute
class Truck(Vehicle):
    def __init__(self,brand, model,year):
        super().__init__(brand, model)
        self.year = year

newCar = Car("Toyota", "Corolla")

print(newCar.brand) # Output: Toyota
print(newCar.model) # Output: Corolla

newTruck = Truck("Toyota", "Hilux", 2020)

print(newTruck.brand) # Output: Toyota
print(newTruck.model) # Output: Hilux
print(newTruck.year) # Output: 2020
